backend/tests/v5/unit/middleware/dataConverter/inputs/teamspaces/settings.test.js

```diff
@@ -433,9 +437,115 @@ const testCheckTicketTemplateExists = () => {
 
 			expect(TemplateModelSchema.getTemplateById).toHaveBeenCalledWith(req.params.teamspace, req.params.template);
 			expect(next).not.toHaveBeenCalled();
-			expect(req.templiateData).toBeUndefined();
+			expect(req.templateData).toBeUndefined();
+			expect(Responder.respond).toHaveBeenCalledWith(
+				req, res, templates.templateNotFound,
+			);
+		});
+	});
+};
+
+const testCheckTicketExists = () => {
+	describe('Check if ticket exists', () => {
+		test('Should call next if the ticket exists', async () => {
+			const template = {
+				type: generateRandomString(),
+				[generateRandomString()]: generateRandomString(),
+			};
+			const ticket = {
+				type: generateRandomString(),
+				name: generateRandomString(),
+				[generateRandomString()]: generateRandomString(),
+			};
+			TemplateModelSchema.getTemplateById.mockResolvedValueOnce(template);
+			TicketModelSchema.getTicketById.mockResolvedValueOnce(ticket);
+
+			const req = {
+				params: {
+					teamspace: generateRandomString(),
+					container: generateRandomString(),
+					project: generateRandomString(),
+					ticket: generateRandomString(),
+				},
+			};
+
+			const res = {};
+			const next = jest.fn();
+
+			await TeamspaceSettings.checkTicketExists(req, res, next);
+
+			expect(TemplateModelSchema.getTemplateById).toHaveBeenCalledWith(req.params.teamspace, ticket.type,
+				{ properties: 1, modules: 1, config: 1 });
+			expect(TicketModelSchema.getTicketById).toHaveBeenCalledWith(req.params.teamspace, req.params.project,
+				req.params.container, req.params.ticket, { type: 1, modules: 1, properties: 1 });
+			expect(next).toHaveBeenCalledTimes(1);
+			expect(req.templateData).toEqual(template);
+			expect(req.ticketData).toEqual(ticket);
+			expect(Responder.respond).not.toHaveBeenCalled();
+		});
+
+		test(`Should respond with ${templates.ticketNotFound} if ticket does not exist`, async () => {
```

Comment:
pretty sure we need .code? ticketNotfound would be an object and it'd probably show up as `{object object}`

```suggestion
		test(`Should respond with ${templates.ticketNotFound.code} if ticket does not exist`, async () => {
```